src/sentry/api/endpoints/project_codeowners.py

```diff
@@ -48,7 +49,13 @@ def validate(self, attrs):
         teamnames, usernames, emails = parse_code_owners(attrs["raw"])
 
         # Check if there exists Sentry users with the emails listed in CODEOWNERS
-        user_emails = UserEmail.objects.filter(email__in=emails)
+        user_emails = UserEmail.objects.annotate(
+            user_in_org=Exists(
+                OrganizationMember.objects.filter(
+                    user=OuterRef("user"), organization=self.context["project"].organization
+                )
+            )
+        ).filter(email__in=emails, user_in_org=True)
```

Comment:
@NisanthanNanthakumar I'm not too familiar with `annotate` and `OuterRef` so I just want to make sure I'm understanding this query correctly.

The `OuterRef("user")` gets evaluated after we've filtered through the `filter(email__in=emails,` part? I think I'm confused on [this part of the docs](https://docs.djangoproject.com/en/3.1/ref/models/expressions/#django.db.models.OuterRef), and how the outer query could be resolved if we have `user_in_org=True` as part of the outer query.